### PR TITLE
ELEMWISE 1arg_2types add MKL kernels

### PR DESCRIPTION
**Pull request by densmirn**
Not all the operations supported by MKL, so separated table of the operations on 2 ones (supported and unsupported by MKL).
`cl::sycl::fabs` is equal to `oneapi::mkl::vm::abs`
`cl::sycl::log` is equal to `oneapi::mkl::vm::ln`

### CODE REVIEW

File: dpnp/backend/custom_1arg_2type_mkl_tbl.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(log2, cl::sycl::log2, oneapi::mkl::vm::log2)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(log, cl::sycl::log, oneapi::mkl::vm::ln)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(sin, cl::sycl::sin, oneapi::mkl::vm::sin)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(sinh, cl::sycl::sinh, oneapi::mkl::vm::sinh)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(sqrt, cl::sycl::sqrt, oneapi::mkl::vm::sqrt)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(tan, cl::sycl::tan, oneapi::mkl::vm::tan)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(tanh, cl::sycl::tanh, oneapi::mkl::vm::tanh)
+MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(trunc, cl::sycl::trunc, oneapi::mkl::vm::trunc)
+
+#undef MACRO_CUSTOM_1ARG_2TYPES_MKL_OP
```
Comment (shssf) [inline dpnp/backend/custom_1arg_2type_mkl_tbl.hpp:67]: Please keep all these things in the same files (do not rename file)

File: dpnp/backend/custom_kernels_elemwise.cpp [hunk 2]
```diff
@@ -46,7 +46,7 @@
             size_t i = global_id[0]; /*for (size_t i = 0; i < size; ++i)*/                                             \
             {                                                                                                          \
                 _DataType_output input_elem = array1[i];                                                               \
-                result[i] = __operation__;                                                                             \
+                result[i] = __operation__(input_elem);                                                                 \
```
Comment (shssf) [inline dpnp/backend/custom_kernels_elemwise.cpp:49]: please keep `input_elem` as before. I would propose to keep this flexibility for this moment.

File: dpnp/backend/custom_kernels_elemwise.cpp [hunk 3]
```diff
@@ -67,6 +67,50 @@
 
 #include <custom_1arg_2type_tbl.hpp>
 
+#define MACRO_CUSTOM_1ARG_2TYPES_MKL_OP(__name__, __operation__, __mkl_operation__)                                    \
```
Comment (shssf) [inline dpnp/backend/custom_kernels_elemwise.cpp:70]: - Could you please add documentation for this macro (if is not exists).
- please do not use "MKL" in names

File: dpnp/backend/backend_iface.hpp [hunk 4]
```diff
@@ -400,6 +400,10 @@ INP_DLLEXPORT void custom_var_c(
     template <typename _DataType_input, typename _DataType_output>                                                     \
     INP_DLLEXPORT void custom_elemwise_##__name__##_c(void* array1, void* result1, size_t size);
 
+#define MACRO_CUSTOM_1ARG_2TYPES_2OPS(__name__, __operation1__, __operation2__)                                        \
```
Comment (shssf) [inline dpnp/backend/backend_iface.hpp:403]: Why we need extra macro? It is the same as existed `MACRO_CUSTOM_1ARG_2TYPES_OP`

Comment (densmirn) [reply dpnp/backend/backend_iface.hpp]: Done!

**Review by shssf — APPROVED** [review]
👍

**Comment by shssf** [comment]
I hope I resolved sources conflict correctly.

**Comment by densmirn** [comment]
> I hope I resolved sources conflict correctly.

Looks like yes.

**Pull request merged**